docs/decisions/0003-instructor-dashboard-rest-api-architecture.rst

```diff
@@ -0,0 +1,266 @@
++--------+--------------------------------------------------------------------+--------------------------------+-----------+-------------------------------------------------------------------+----------------------------------------------------------------------------------------------+
+| POST   | /api/instructor/v1/courses/{course_key}/enrollments/csv_upload     | bulk_enroll, util              | Missing   | N/A                                                               | Upload CSV file to bulk enroll/unenroll students                                             |
++--------+--------------------------------------------------------------------+--------------------------------+-----------+-------------------------------------------------------------------+----------------------------------------------------------------------------------------------+
+| GET    | /api/instructor/v1/courses/{course_key}/roles/{role_type}/members  | student.auth                   | Partial   | /courses/{course_id}/instructor/api/list_course_role_members      | List users with a specific course role (staff, instructor, beta_tester, etc.)                |
++--------+--------------------------------------------------------------------+--------------------------------+-----------+-------------------------------------------------------------------+----------------------------------------------------------------------------------------------+
+| POST   | /api/instructor/v1/courses/{course_key}/roles/{username}           | student.auth                   | Partial   | /courses/{course_id}/instructor/api/modify_access                 | Grant a course role to a user (role specified in request body)                               |
++--------+--------------------------------------------------------------------+--------------------------------+-----------+-------------------------------------------------------------------+----------------------------------------------------------------------------------------------+
+| DELETE | /api/instructor/v1/courses/{course_key}/roles/{username}           | student.auth                   | Missing   | N/A                                                               | Revoke a course role from a user (role specified in request body)                            |
++--------+--------------------------------------------------------------------+--------------------------------+-----------+-------------------------------------------------------------------+----------------------------------------------------------------------------------------------+
+| GET    | /api/instructor/v1/courses/{course_key}/beta_testers               | student.auth                   | Missing   | N/A                                                               | List all beta testers for the course                                                         |
```

Comment:
The general `roles` endpoint already covers beta testers, so why do they need their own endpoint? Same for forum members - wouldn't a generic "roles" endpoint cover both of these use cases already?